### ISSUE TITLE
How to use the bbox dataset on yolo？

### ISSUE THREAD

**Issue by leevinci** [ultralytics/yolov5#6253]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I have a dataset marked with bbox, but yolo uses txt, I have tried many programs to convert it, but all failed. I'm still trying to learn python, so I would like to ask if there is a solution, or if there are related cases that can help 

{"cju0qkwl35piu0993l0dewei2": {"height": 529, "width": 622, "bbox": [{"label": "polyp", "xmin": 38, "ymin": 5, "xmax": 430, "ymax": 338}]}, "cju0qoxqj9q6s0835b43399p4": {"height": 1070, "width": 1348, "bbox": [{"label": "polyp", "xmin": 194, "ymin": 284, "xmax": 913, "ymax": 1049}]}, "cju0qx73cjw570799j4n5cjze": {"height": 529, "width": 619, "bbox": [{"label": "polyp", "xmin": 187, "ymin": 14, "xmax": 543, "ymax": 526}]}, "cju0roawvklrq0799vmjorwfv": {"height": 528, "width": 622, "bbox": [{"label": "polyp", "xmin": 486, "ymin": 42, "xmax": 622, "ymax": 339}, {"label": "polyp", "xmin": 329, "ymin": 71, "xmax": 437, "ymax": 150}, {"label": "polyp", "xmin": 619, "ymin": 151, "xmax": 622, "ymax": 285}]}, "cju0rx1idathl0835detmsp84": {"height": 528, "width": 617, "bbox": [{"label": "polyp", "xmin": 292, "ymin": 154, "xmax": 481, "ymax": 375}]}, "cju0s2a9ekvms080138tjjpxr": {"height": 547, "width": 626, "bbox": [{"label": "polyp", "xmin": 379, "ymin": 57, "xmax": 626, "ymax": 431}]}, "cju0s690hkp960855tjuaqvv0": {"height": 530, "width": 621, "bbox": [{"label": "polyp", "xmin": 223, "ymin": 285, "xmax": 336, "ymax": 434}]}, "cju0sr5ghl0nd08789uzf1raf": {"height": 1019, "width": 1214, "bbox": [{"label": "polyp", "xmin": 393, "ymin": 178, "xmax": 730, "ymax": 488}, {"label": "polyp", "xmin": 300, "ymin": 571, "xmax": 698, "ymax": 975}]}, "cju0sxqiclckk08551ycbwhno": {"height": 531, "width": 570, "bbox": [{"label": "polyp", "xmin": 400, "ymin": 278, "xmax": 543, "ymax": 416}]}, "cju0t4oil7vzk099370nun5h9": {"height": 530, "width": 622, "bbox": [{"label": "polyp", "xmin": 203, "ymin": 115, "xmax": 573, "ymax": 526}]}, "cju0tl3uz8blh0993wxvn7ly3": {"height": 529, "width": 622, "bbox": [{"label": "polyp", "xmin": 314, "ymin": 386, "xmax": 370, "ymax": 435}]}, "cju0u2g7pmnux0801vkk47ivj": {"height": 530, "width": 611, "bbox": [{"label": "polyp", "xmin": 0, "ymin": 152, "xmax": 367, "ymax": 530}]}, "cju0u82z3cuma0835wlxrnrjv": {"height": 546, "width": 626, "bbox": [{"label": "polyp", "xmin": 293, "ymin": 90, "xmax": 609, "ymax": 531}]}, "cju5k7r0yf98c09878csbxb4d": {"height": 528, "width": 622, "bbox": [{"label": "polyp", "xmin": 228, "ymin": 40, "xmax": 321, "ymax": 134}]}, "cju5klveuff6w0871wbibgh3m": {"height": 530, "width": 574, "bbox": [{"label": "polyp", "xmin": 185, "ymin": 228, "xmax": 308, "ymax": 350}]}, "cju5knbbqfipk080128cggukq": {"height": 529, "width": 597, "bbox": [{"label": "polyp", "xmin": 194, "ymin": 68, "xmax": 595, "ymax": 524}]}}

**Comment by glenn-jocher** [comment]
@1235li 👋 Hello! Thanks for asking about YOLOv5 🚀 **dataset formatting**. To train correctly your data must be in YOLOv5 format. Please see our [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) tutorial for full documentation on dataset setup and all steps required to start training your first model. A few excerpts from the tutorial:

### 1.1 Create dataset.yaml

[COCO128](https://www.kaggle.com/datasets/ultralytics/coco128) is an example small tutorial dataset composed of the first 128 images in [COCO](http://cocodataset.org/#home) train2017. These same 128 images are used for both training and validation to verify our training pipeline is capable of overfitting. [data/coco128.yaml](https://github.com/ultralytics/yolov5/blob/master/data/coco128.yaml), shown below, is the dataset config file that defines 1) the dataset root directory `path` and relative paths to `train` / `val` / `test` image directories (or *.txt files with image paths), 2) the number of classes `nc` and 3) a list of class `names`:
```yaml
# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path: ../datasets/coco128  # dataset root dir
train: images/train2017  # train images (relative to 'path') 128 images
val: images/train2017  # val images (relative to 'path') 128 images
test:  # test images (optional)

# Classes
nc: 80  # number of classes
names: [ 'person', 'bicycle', 'car', 'motorcycle', 'airplane', 'bus', 'train', 'truck', 'boat', 'traffic light',
         'fire hydrant', 'stop sign', 'parking meter', 'bench', 'bird', 'cat', 'dog', 'horse', 'sheep', 'cow',
         'elephant', 'bear', 'zebra', 'giraffe', 'backpack', 'umbrella', 'handbag', 'tie', 'suitcase', 'frisbee',
         'skis', 'snowboard', 'sports ball', 'kite', 'baseball bat', 'baseball glove', 'skateboard', 'surfboard',
         'tennis racket', 'bottle', 'wine glass', 'cup', 'fork', 'knife', 'spoon', 'bowl', 'banana', 'apple',
         'sandwich', 'orange', 'broccoli', 'carrot', 'hot dog', 'pizza', 'donut', 'cake', 'chair', 'couch',
         'potted plant', 'bed', 'dining table', 'toilet', 'tv', 'laptop', 'mouse', 'remote', 'keyboard', 'cell phone',
         'microwave', 'oven', 'toaster', 'sink', 'refrigerator', 'book', 'clock', 'vase', 'scissors', 'teddy bear',
         'hair drier', 'toothbrush' ]  # class names
```


### 1.2 Create Labels

After using a tool like [Roboflow Annotate](https://roboflow.com/annotate?ref=ultralytics) to label your images, export your labels to **YOLO format**, with one `*.txt` file per image (if no objects in image, no `*.txt` file is required). The `*.txt` file specifications are:

- One row per object
- Each row is `class x_center y_center width height` format.
- Box coordinates must be in **normalized xywh** format (from 0 - 1). If your boxes are in pixels, divide `x_center` and `width` by image width, and `y_center` and `height` by image height.
- Class numbers are zero-indexed (start from 0).

<img width="800" alt="Image Labels" src="https://user-images.githubusercontent.com/26833433/91506361-c7965000-e886-11ea-8291-c72b98c25eec.jpg">

The label file corresponding to the above image contains 2 persons (class `0`) and a tie (class `27`):

<p align="center"><img width="428" src="https://user-images.githubusercontent.com/26833433/112467037-d2568c00-8d66-11eb-8796-55402ac0d62f.png"></p>


### 1.3 Organize Directories

Organize your train and val images and labels according to the example below. YOLOv5 assumes  `/coco128` is inside a `/datasets` directory **next to** the `/yolov5` directory. **YOLOv5 locates labels automatically for each image** by replacing the last instance of `/images/` in each image path with `/labels/`. For example:
```bash
../datasets/coco128/images/im0.jpg  # image
../datasets/coco128/labels/im0.txt  # label
```

<p align="center"><img width="700" src="https://user-images.githubusercontent.com/26833433/134436012-65111ad1-9541-4853-81a6-f19a3468b75f.png"></p>
</details>


Good luck 🍀 and let us know if you have any other questions!
